formats/nested_type_param.ksy

```diff
@@ -1,18 +1,15 @@
 meta:
-  id: test
-  endian: be
+  id: nested_type_param
 seq:
-  - id: seq_block
-    type: test::my_type(2)
+  - id: main_seq
+    type: nested_type_param::my_type(5)
 types:
   my_type:
     params:
-      - id: repeat_count
-        type: s4
+      - id: my_len
+        type: u4
     seq:
-      - id: world
-        type: s4
-      - id: repeated_thing
-        type: s4
-        repeat: expr
-        repeat-expr: repeat_count
+      - id: body
+        type: str
+        size: my_len
+        encoding: UTF-8
```

Comment:
It's definitely better, but it could be even simpler:

```yaml
meta:
  id: nested_type_param
seq:
  - id: foo
    type: nested_type_param::my_type(5)
types:
  my_type:
    params:
      - id: int_param
        type: u4
    seq: [] # this line can be just omitted
```

My point is, that it's completely fine to leave the `seq` list empty, or leave it out completely, because it's not vital for the test to work.

But I have a unpleasant feeling of being a pedant here, the thing that you've done is fine too, just a little bit more complex than it has to be. If you're satisfied with it, me too. It just seems to me that you don't know it's possible to leave the `seq` empty, but if you feel safer when it actually parses something from the stream, it's OK.